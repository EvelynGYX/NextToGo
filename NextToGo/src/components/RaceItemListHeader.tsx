import React from 'react';
import {StyleSheet, View} from 'react-native';
import FontAwesome5Icon from 'react-native-vector-icons/FontAwesome5';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import Category from '../constants/Category';
import {IFilter} from '../interfaces/Filter';

interface RaceItemListHeaderProps extends IFilter {}

const RaceItemListHeader = (props: RaceItemListHeaderProps) => {
  const isChecked = (isSelected: boolean): string => {
    if (isSelected) return 'check-circle';
    else return 'circle';
  };

  return (
    <View style={styles.container}>
      <FontAwesome5Icon
        style={styles.icon}
        name={isChecked(props.categorySelected.horseSelected)}
        onPress={() => props.onCategorySelected(Category.Horse)}
        size={20}
      />
      <FontAwesome5Icon style={styles.icon} name="horse-head" size={20} />
      <FontAwesome5Icon
        style={styles.icon}
        name={isChecked(props.categorySelected.greyHoundSelected)}
        onPress={() => props.onCategorySelected(Category.Greyhound)}
        size={20}
      />
      <FontAwesome5Icon style={styles.icon} name="dog" size={20} />
      <FontAwesome5Icon
        style={styles.icon}
        name={isChecked(props.categorySelected.harnessSelected)}
        onPress={() => props.onCategorySelected(Category.Harness)}
        size={20}
      />
      <MaterialCommunityIcons
        style={styles.icon}
        name="horse-human"
        size={20}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignContent: 'center',
    justifyContent: 'flex-start',
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    paddingHorizontal: 12,
    paddingVertical: 16,
  },
  icon: {
    paddingRight: 8,
  },
});

export default RaceItemListHeader;
