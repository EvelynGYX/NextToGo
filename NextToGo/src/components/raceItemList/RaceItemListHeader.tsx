import React from 'react';
import {StyleSheet, View} from 'react-native';
import Category from '../../constants/Category';
import {IFilter} from '../../interfaces/Filter';
import CheckIcon, {CheckIconProps} from '../icons/CheckIcon';
import RaceIcon from '../icons/RaceIcon';

export type RaceItemListHeaderProps = IFilter;

interface IHeaderConfig extends CheckIconProps {
  category: Category;
}

const RaceItemListHeader = (props: RaceItemListHeaderProps) => {
  const headerConfigs: IHeaderConfig[] = [
    {
      isChecked: props.categorySelected.horseSelected,
      onPress: () => props.onCategorySelected(Category.Horse),
      category: Category.Horse,
    },
    {
      isChecked: props.categorySelected.greyHoundSelected,
      onPress: () => props.onCategorySelected(Category.Greyhound),
      category: Category.Greyhound,
    },
    {
      isChecked: props.categorySelected.harnessSelected,
      onPress: () => props.onCategorySelected(Category.Harness),
      category: Category.Harness,
    },
  ];

  return (
    <View style={styles.container}>
      {headerConfigs.map((config, index) => (
        <View style={styles.content} key={index}>
          <CheckIcon
            isChecked={config.isChecked}
            onPress={config.onPress}
            style={styles.icon}
          />
          <RaceIcon category={config.category} style={styles.icon} />
        </View>
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignContent: 'center',
    justifyContent: 'flex-start',
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    paddingHorizontal: 12,
    paddingVertical: 16,
  },
  content: {
    flexDirection: 'row',
  },
  icon: {
    paddingRight: 8,
  },
});

export default RaceItemListHeader;
