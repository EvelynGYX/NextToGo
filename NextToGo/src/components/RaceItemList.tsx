import React from 'react';
import {FlatList, StyleSheet, View} from 'react-native';
import {IFilter} from '../interfaces/Filter';
import {RaceArray} from '../interfaces/RaceArrayType';
import RaceItem from './RaceItem';
import RaceItemListDivider from './RaceItemListDivider';
import RaceItemListHeader from './RaceItemListHeader';

interface RaceItemListProps extends IFilter {
  data: RaceArray;
}

const RaceItemList = (props: RaceItemListProps) => {
  return (
    <View style={styles.contentWrapper}>
      <FlatList
        data={props.data}
        keyExtractor={item => item.raceId}
        renderItem={({item}) => <RaceItem {...item} />}
        ItemSeparatorComponent={RaceItemListDivider}
        ListHeaderComponent={() => (
          <RaceItemListHeader {...(props as IFilter)} />
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  contentWrapper: {
    borderTopWidth: 1,
    borderBottomWidth: 1,
    borderColor: 'grey',
  },
});

export default RaceItemList;
