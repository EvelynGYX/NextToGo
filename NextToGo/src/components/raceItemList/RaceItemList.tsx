import React from 'react';
import {FlatList, StyleSheet, Text, View} from 'react-native';
import {IFilter} from '../../interfaces/Filter';
import {RaceArray} from '../../interfaces/RaceArray';
import LoadingScreen from '../../screens/LoadingScreen';
import RaceItem from './RaceItem';
import RaceItemListDivider from './RaceItemListDivider';
import RaceItemListHeader from './RaceItemListHeader';

export interface RaceItemListProps extends IFilter {
  data: RaceArray;
  loading: boolean;
}

const RaceItemList = (props: RaceItemListProps) => {
  if (props.loading || props.data.length === 0) {
    return (
      <View style={styles.container}>
        <RaceItemListHeader {...(props as IFilter)} />
        {props.loading ? (
          <LoadingScreen />
        ) : (
          <View style={styles.textContainer}>
            <Text>No race</Text>
          </View>
        )}
      </View>
    );
  }

  return (
    <View style={styles.contentWrapper}>
      <FlatList
        data={props.data}
        keyExtractor={item => item.raceId}
        renderItem={({item}) => <RaceItem {...item} />}
        ItemSeparatorComponent={RaceItemListDivider}
        ListHeaderComponent={() => (
          <RaceItemListHeader {...(props as IFilter)} />
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '100%',
    borderTopWidth: 1,
    borderColor: 'grey',
  },
  contentWrapper: {
    borderTopWidth: 1,
    borderBottomWidth: 1,
    borderColor: 'grey',
  },
  textContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default React.memo(RaceItemList);
