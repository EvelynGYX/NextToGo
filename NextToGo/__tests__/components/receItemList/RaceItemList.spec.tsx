import {cleanup} from '@testing-library/react-hooks';
import {shallow} from 'enzyme';
import React from 'react';
import {FlatList, Text} from 'react-native';
import renderer from 'react-test-renderer';
import RaceItemList, {
  RaceItemListProps,
} from '../../../src/components/raceItemList/RaceItemList';
import RaceItemListHeader from '../../../src/components/raceItemList/RaceItemListHeader';
import Category from '../../../src/constants/Category';
import LoadingScreen from '../../../src/screens/LoadingScreen';

describe('RaceItemList', () => {
  const race = {
    raceId: 'some race id',
    meetingName: 'meeting name',
    raceNumber: 'race number' as any as number,
    category: Category.Greyhound,
    advertisedStart: 'some advertisedStart',
    venueState: 'some venueState',
  };
  let mockProps: RaceItemListProps = {
    onCategorySelected: 'some categorySelected' as any as (
      category: Category,
    ) => void,
    categorySelected: {
      horseSelected: 'some horseSelected' as any as boolean,
      greyHoundSelected: 'some greyHoundSelected' as any as boolean,
      harnessSelected: 'some harnessSelected' as any as boolean,
    },
    data: [race],
    loading: 'some loading' as any as boolean,
  };

  afterEach(cleanup);
  it('renders correctly if data prop is not empty', () => {
    mockProps.loading = false;

    const tree = renderer.create(<RaceItemList {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if data prop is empty', () => {
    mockProps.loading = false;
    mockProps.data = [];

    const tree = renderer.create(<RaceItemList {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if loading prop is true', () => {
    mockProps.loading = true;

    const tree = renderer.create(<RaceItemList {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders No race with RaceItemListHeader if data prop is an empty array', () => {
    mockProps.loading = false;
    mockProps.data = [];

    const wrapper = shallow(<RaceItemList {...mockProps} />);
    expect(wrapper.find(Text).props().children).toEqual('No race');
    expect(wrapper.exists(RaceItemListHeader)).toBe(true);
    expect(wrapper.exists(FlatList)).toBe(false);
    expect(wrapper.exists(LoadingScreen)).toBe(false);
  });

  it('renders Flatlist if data prop is not an empty array', () => {
    mockProps.loading = false;
    mockProps.data = [race];

    const wrapper = shallow(<RaceItemList {...mockProps} />);
    expect(wrapper.exists(FlatList)).toBe(true);
    expect(wrapper.exists(RaceItemListHeader)).toBe(false);
    expect(wrapper.exists(LoadingScreen)).toBe(false);
  });

  it('renders LoadingScreen with RaceItemListHeader if loading prop is true', () => {
    mockProps.loading = true;
    const wrapper = shallow(<RaceItemList {...mockProps} />);
    expect(wrapper.exists(LoadingScreen)).toBe(true);
    expect(wrapper.exists(FlatList)).toBe(false);
    expect(wrapper.exists(RaceItemListHeader)).toBe(true);
  });
});
