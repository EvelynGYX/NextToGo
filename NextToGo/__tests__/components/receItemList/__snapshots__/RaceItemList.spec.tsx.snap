// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaceItemList renders correctly if data prop is empty 1`] = `
<View
  style={
    Object {
      "borderColor": "grey",
      "borderTopWidth": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "borderBottomColor": "grey",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingHorizontal": 12,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Regular",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "400",
            },
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Regular",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "400",
            },
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Regular",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "400",
            },
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󱗀
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      No race
    </Text>
  </View>
</View>
`;

exports[`RaceItemList renders correctly if data prop is not empty 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 1,
      "borderColor": "grey",
      "borderTopWidth": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={
      Array [
        Object {
          "advertisedStart": "some advertisedStart",
          "category": "9daef0d7-bf3c-4f50-921d-8e818c60fe61",
          "meetingName": "meeting name",
          "raceId": "some race id",
          "raceNumber": "race number",
          "venueState": "some venueState",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingHorizontal": 12,
              "paddingVertical": 16,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 20,
                  },
                  Object {
                    "paddingRight": 8,
                  },
                  Object {
                    "fontFamily": "FontAwesome5Free-Regular",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "400",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 20,
                  },
                  Object {
                    "paddingRight": 8,
                  },
                  Object {
                    "fontFamily": "FontAwesome5Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 20,
                  },
                  Object {
                    "paddingRight": 8,
                  },
                  Object {
                    "fontFamily": "FontAwesome5Free-Regular",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "400",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 20,
                  },
                  Object {
                    "paddingRight": 8,
                  },
                  Object {
                    "fontFamily": "FontAwesome5Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 20,
                  },
                  Object {
                    "paddingRight": 8,
                  },
                  Object {
                    "fontFamily": "FontAwesome5Free-Regular",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "400",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 20,
                  },
                  Object {
                    "paddingRight": 8,
                  },
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󱗀
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingHorizontal": 12,
              "paddingVertical": 8,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 20,
                  },
                  Object {
                    "paddingRight": 24,
                  },
                  Object {
                    "fontFamily": "FontAwesome5Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
            <View
              style={
                Object {
                  "flexDirection": "column",
                }
              }
            >
              <Text>
                meeting name Rrace number
              </Text>
              <Text>
                some venueState
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text>
              some advertisedStart
            </Text>
            <View
              style={
                Object {
                  "paddingLeft": 4,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 18,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Entypo",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`RaceItemList renders correctly if loading prop is true 1`] = `
<View
  style={
    Object {
      "borderColor": "grey",
      "borderTopWidth": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "borderBottomColor": "grey",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingHorizontal": 12,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Regular",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "400",
            },
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Regular",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "400",
            },
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Regular",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "400",
            },
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "paddingRight": 8,
            },
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󱗀
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      size="large"
    />
  </View>
</View>
`;
